src/studio/src/designer/backend/Services/Implementation/SchemaModelService.cs

```diff
@@ -68,26 +89,106 @@ public async Task UpdateSchema(string org, string repository, string developer,
         {
             var altinnGitRepository = _altinnGitRepositoryFactory.GetAltinnGitRepository(org, repository, developer);
 
-            if (altinnGitRepository.GetRepositoryType().Result == Enums.AltinnRepositoryType.App)
+            if (await altinnGitRepository.GetRepositoryType() == AltinnRepositoryType.App)
             {
                 await UpdateAllAppModelFiles(org, repository, developer, relativeFilePath, jsonContent);
             }
             else 
             {              
                 await altinnGitRepository.WriteTextByRelativePathAsync(relativeFilePath, jsonContent, true);
-            } 
+            }
+        }
+
+        /// <summary>
+        /// Builds a JSON schema based on the uploaded XSD.
+        /// </summary>
+        /// <remarks>
+        /// This operation is using the new data modelling library.
+        /// </remarks>
+        /// <param name="org">Organization owning the repository identified by it's short name.</param>
+        /// <param name="repository">Repository name to search for schema files.</param>
+        /// <param name="developer">Developers short name</param>
+        /// <param name="fileName">The name of the file being uploaded.</param>
+        /// <param name="xsdStream">Stream representing the XSD.</param>
+        public async Task<string> BuildSchemaFromXsd(
```

Comment:
Could we split this into smaller methods? I was thinking let this method orchestrate but separate methods for the 3 cases? Datamodels, App (Old, New)

Reply:
I moved a part of it to a separate method, but left most of it as is.